core/src/main/java/org/eclipse/hono/util/CredentialsObject.java

```diff
@@ -369,17 +367,16 @@ public static JsonObject emptySecret(final Instant notBefore, final Instant notA
      * @param salt The salt to prepend the password bytes with.
      * @param password The password to hash.
      * @return The hashed password.
-     * @throws NoSuchAlgorithmException if the given hash function is not supported on
-     *           the JVM.
      */
-    public static byte[] getHashedPassword(final String hashFunction, final byte[] salt, final String password) throws NoSuchAlgorithmException {
 
-        final MessageDigest digest = MessageDigest.getInstance(hashFunction);
+    public static String getHashedPassword(final String hashFunction, final byte[] salt, final String password) {
+        //TODO Delegating encoder support encoding only using default encoder. Maybe we can enhace the interface to support more hash functions
+        final PasswordEncoder encoder = new MessageDigestPasswordEncoder(hashFunction);
```

Comment:
why aren't you using the `PasswordEncoderFactory` that you have created in the same package?